sql-plugin/src/main/scala/com/nvidia/spark/RebaseHelper.scala

```diff
@@ -37,9 +36,9 @@ object RebaseHelper extends Arm {
         }
       }
     } else if (dtype.isTimestamp) {
-      assert(dtype == DType.TIMESTAMP_MICROSECONDS)
+      assert(dtype == DType.TIMESTAMP_MICROSECONDS || dtype == DType.TIMESTAMP_MILLISECONDS)
       withResource(
-        Scalar.timestampFromLong(DType.TIMESTAMP_MICROSECONDS, startTs)) { minGood =>
+        Scalar.timestampFromLong(column.getDataType, startTs)) { minGood =>
```

Comment:
`startTs` was calculated in microseconds.  Just saying that it is now milliseconds does not make it so.

Reply:
That was stupid. I have updated it. :facepalm: